pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1226 @@
+			return err
+		}
+	}
+
+	if g.Example != nil {
+		if err := g.Example.ToRawInfo().Decode(&k.Example); err != nil {
+			return err
+		}
+	}
+
+	// SchemaProps
+	if err := k.Ref.FromGnostic(g.XRef); err != nil {
+		return err
+	}
+	k.Type = g.Type.GetValue()
+	k.Format = g.GetFormat()
+	k.Title = g.GetTitle()
+
+	// These below fields are not available in gnostic types, so will never
+	// be populated. This means roundtrips which make use of these
+	//	(non-official, kube-only) fields will lose information.
+	//
+	// Schema.ID is not available in official spec
+	// Schema.$schema
+	// Schema.Nullable - in openapiv3, not v2
+	// Schema.AnyOf - in openapiv3, not v2
+	// Schema.OneOf - in openapiv3, not v2
+	// Schema.Not - in openapiv3, not v2
+	// Schema.PatternProperties - in openapiv3, not v2
+	// Schema.Dependencies - in openapiv3, not v2
+	// Schema.AdditionalItems
+	// Schema.Definitions - not part of spec
+	// Schema.ExtraProps - gnostic parser rejects any keys it does not recognize
+
+	if g.GetDefault() != nil {
+		if err := g.GetDefault().ToRawInfo().Decode(&k.Default); err != nil {
+			return err
+		}
+	}
+
+	// These conditionals (!= 0) follow gnostic's logic for ToRawInfo
+	// The keys in gnostic source are only included if nonzero.
+
+	if g.Maximum != 0.0 {
+		k.Maximum = &g.Maximum
+	}
+
+	if g.Minimum != 0.0 {
+		k.Minimum = &g.Minimum
+	}
+
+	k.ExclusiveMaximum = g.ExclusiveMaximum
+	k.ExclusiveMinimum = g.ExclusiveMinimum
+
+	if g.MaxLength != 0 {
+		k.MaxLength = &g.MaxLength
+	}
+
+	if g.MinLength != 0 {
+		k.MinLength = &g.MinLength
+	}
+
+	k.Pattern = g.GetPattern()
+
+	if g.MaxItems != 0 {
+		k.MaxItems = &g.MaxItems
+	}
+
+	if g.MinItems != 0 {
+		k.MinItems = &g.MinItems
+	}
+	k.UniqueItems = g.UniqueItems
+
+	if g.MultipleOf != 0 {
+		k.MultipleOf = &g.MultipleOf
+	}
+
+	for _, v := range g.GetEnum() {
+		var convert interface{}
+		if err := v.ToRawInfo().Decode(&convert); err != nil {
+			return err
+		}
+		k.Enum = append(k.Enum, convert)
+	}
+
+	if g.MaxProperties != 0 {
+		k.MaxProperties = &g.MaxProperties
+	}
+
+	if g.MinProperties != 0 {
+		k.MinProperties = &g.MinProperties
+	}
+
+	k.Required = g.Required
+
+	if g.GetItems() != nil {
+		k.Items = &SchemaOrArray{}
+		if len(g.Items.GetSchema()) == 1 {
+			k.Items.Schema = &Schema{}
+			if err := k.Items.Schema.FromGnostic(g.Items.Schema[0]); err != nil {
+				return err
+			}
+		} else {
+			for _, v := range g.Items.Schema {
+				schema := Schema{}
+				if err := schema.FromGnostic(v); err != nil {
+					return nil
+				}
+				k.Items.Schemas = append(k.Items.Schemas, schema)
+			}
+		}
+	}
+
+	for i, v := range g.GetAllOf() {
+		k.AllOf = append(k.AllOf, Schema{})
+		if err := k.AllOf[i].FromGnostic(v); err != nil {
+			return err
+		}
+	}
+
+	if g.Properties != nil {
+		k.Properties = make(map[string]Schema)
+		for _, namedSchema := range g.Properties.AdditionalProperties {
+			val := &Schema{}
+			if err := val.FromGnostic(namedSchema.Value); err != nil {
```

Comment:
namedSchema can be null, right?

Reply:
null checks in all ranges over loops have been added/verified